core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -305,7 +306,29 @@ private static Handler configureFileHandler(final FileConfig config, final Error
             handler = periodicRotatingFileHandler;
         }
 
-        final PatternFormatter formatter = new PatternFormatter(config.format);
+        // Check if formatter class is set and can be initialized.
+        Formatter formatter = null;
+        if (config.formatter.isPresent()) {
+            try {
+                Class<?> formatterClass = Class.forName(config.formatter.get(), false,
+                        Thread.currentThread().getContextClassLoader());
```

Comment:
I don't recall whether this is happening during static initialization. Can you confirm that this works with native images?

Reply:
It works with native images (tested with Podman on Fedora) if the formatter class is included in the build. I made a test project with native and had to add the class to reflection-config.json to get it to work. (I must admit that I didn't test with native the first time around.)

Reply:
@dmlloyd Did you have a chance to look at this?

Reply:
I don't like requiring the reflection config for this to work.  It seems to me that custom formatters should either be configured as build time configuration, or else they should be done by service loader (for example, we could have a `CustomFormatter` interface with a `getName()` method; at static init, all such classes can be loaded and registered to a map, and the run time configuration would select a formatter by name instead of by class).

Reply:
Thanks for the feedback! I can see that my proposal was not well-aligned with the project goals, and I don't have time now to implement a generic customizable JSON logger. So I'll close this pull request, maybe I'll be back some other time with a generic JSON logging solution.